osquery/tables/system/windows/system_info.cpp

```diff
@@ -45,17 +46,40 @@ QueryData genSystemInfo(QueryContext& context) {
     r["cpu_logical_cores"] = INTEGER(numProcs);
     wmiResultsProc[0].GetLong("NumberOfCores", numProcs);
     r["cpu_physical_cores"] = INTEGER(numProcs);
-    wmiResults[0].GetString("TotalPhysicalMemory", r["physical_memory"]);
     wmiResults[0].GetString("Manufacturer", r["hardware_vendor"]);
     wmiResults[0].GetString("Model", r["hardware_model"]);
   } else {
     r["cpu_logical_cores"] = "-1";
     r["cpu_physical_cores"] = "-1";
-    r["physical_memory"] = "-1";
     r["hardware_vendor"] = "-1";
     r["hardware_model"] = "-1";
   }
 
+  // Physical memory size is reported from Win32_PhysicalMemory. See
+  // TotalPhysicalMemory at
+  // https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-computersystem
+  const WmiRequest wmiSystemReqMem("select Capacity from Win32_PhysicalMemory");
+  const std::vector<WmiResultItem>& wmiResultsMem = wmiSystemReqMem.results();
+  if (wmiResultsMem.empty()) {
+    r["physical_memory"] = "-1";
+  } else {
+    // capacity is documented as being a uint64, but it actually coming back as
+    // a string.
+    unsigned __int64 sum = 0;
```

Comment:
You should be able to use `uint64_t`

Reply:
I'll try. This was cargo culted from elsewhere